carma_wm/src/CARMAWorldModel.cpp

```diff
@@ -1378,6 +1379,61 @@ namespace carma_wm
     return curr_light;
   }
 
+   bool CARMAWorldModel::check_if_seen_before_movement_state(boost::posix_time::ptime min_end_time_dynamic,lanelet::CarmaTrafficSignalState received_state_dynamic,uint16_t mov_id, uint8_t mov_signal_group)
+    {
+
+      if(sim_.traffic_signal_states_[mov_id][mov_signal_group].empty())
+      {
+          return 0;
+      }
+
+      for(auto mov_check:sim_.traffic_signal_states_[mov_id][mov_signal_group])
+      {
+
```

Comment:
It looks like we never remove the old <end_time, state> pair. And it might become too many especially if we are just sitting there and testing. So maybe create temporary vector that saves only "not outdated" states. A.k.a create `std::vector<std::pair<boost::posix_time::ptime, lanelet::CarmaTrafficSignalState>> temp_signal_states` before this for loop. And push_back if `lanelet::time::timeFromSec(ros::Time::now().toSec()) > mov_check.first`. And remember assign the temporary vector back to original to save before returning